reconcile/aws_cloudwatch_log_retention/integration.py

```diff
@@ -0,0 +1,82 @@
+import logging
+import re
+from collections.abc import Callable
+from typing import Optional
+
+from pydantic import BaseModel
+
+from reconcile import queries
+from reconcile.queries import get_aws_accounts
+from reconcile.utils.aws_api import AWSApi
+
+QONTRACT_INTEGRATION = "aws_cloudwatch_log_retention"
+
+
+class AWSCloudwatchLogRetention(BaseModel):
+    name: str
+    acct_uid: str
+    log_regex: str
+    log_retention_day_length: str
+
+
+def get_app_interface_cloudwatch_retention_period(aws_accounts: list) -> list:
+    # aws_accounts = get_aws_accounts(cleanup=True)
+    results = []
+    for aws_acct in aws_accounts:
+        aws_acct_name = aws_acct.get("name")
+        acct_uid = aws_acct.get("uid")
+        if aws_acct.get("cleanup"):
+            for aws_acct_field in aws_acct.get("cleanup"):
+                if aws_acct_field["provider"] == "cloudwatch":
+                    results.append(
+                        AWSCloudwatchLogRetention(
+                            name=aws_acct_name,
+                            acct_uid=acct_uid,
+                            log_regex=aws_acct_field["regex"],
+                            log_retention_day_length=aws_acct_field[
+                                "retention_in_days"
+                            ],
+                        )
+                    )
+    return results
+
+
+def run(dry_run: bool, thread_pool_size: int, defer: Optional[Callable] = None) -> None:
+    aws_accounts = get_aws_accounts(cleanup=True)
+    cloudwatch_cleanup_list = get_app_interface_cloudwatch_retention_period(
+        aws_accounts
+    )
+    for cloudwatch_cleanup_entry in cloudwatch_cleanup_list:
```

Comment:
One aws account may have multiple cloudwatch_cleanup_entry, we don't need to init awsapi and get log_groups for every cloudwatch_cleanup_entry for the same account in this case. Try to loop every account from aws_accounts first, init awsapi and get log_groups there and then loop every cloudwatch_cleanup_entry in that account.